### DISCUSSION TITLE
Pass user context to useSentry

### DISCUSSION THREAD

**Discussion by dakshshah96** [dotansimha/envelop#1918] [Q&A] [answered]
I'm using the `useSentry` plugin on my GraphQL Yoga server to send errors and transactions to Sentry. My user data like the `userId` is stored in the `context` object.

Is there a way I can [set the scope for my Sentry error](https://docs.sentry.io/platforms/node/enriching-events/identify-user/) with the `userId` and other properties present in `context` while using the Sentry plugin?

**Comment by Warxcell** [comment]
with `configureScope` option.

```
      configureScope: (args, scope) => {
        /** @ts-ignore */
        const user = args.contextValue.user;
        scope.setUser(user)
      },
```


but the bigger question is - how to remove the `/** @ts-ignore */` ?!

**Reply by EmrysMyrddin** [reply]
Unfortunately, for now the types are a bit loose here to allow users to use other engines than `graphql-js`. We are working on improving it :-)

Waiting for that, you can workaround by using this trick:

```ts
// Let's say you have defined you context type somewhere like this:
interface Context {
  user: User
}

// You can then cast the context manually using the `as` keyword
configureScope: (args, scope) => {
  const { user } = args.contextValue as Context;
  scope.setUser(user)
},
```

**Reply by EmrysMyrddin** [reply]
Fix in progress #2103

**Reply by EmrysMyrddin** [reply]
With this fix, you will be able to configure your scope with a typed context:

```ts
interface Context {
  user: User
}

useSentry<Context>({
  // other options ...
  configureScope(args, scope) {
    scope.setUser(args.contextValue.user) // TS should now be happy with this :-)
  }
})

**Reply by EmrysMyrddin** [reply, marked as answer]
Fix released, please update your plugin to `@envelop/sentry@8.0.0` :-)
